### PR TITLE
header tests

### PR DESCRIPTION
**Pull request by nadiyafomenko**

### CODE REVIEW

File: src/redux/header/tests/header.sagas.test.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  getHeaderById,
+  updateHeader
+} from '../header.operations';
+
+import { headers, id, mockHeader } from './header.variables';
+import { config } from '../../../configs';
+import { handleSuccessSnackbar } from '../../snackbar/snackbar.sagas';
+
+import {
+  deleteHeader,
```
Comment (ivan-bonk) [inline src/redux/header/tests/header.sagas.test.js:26]: you shouldn't use this function in this test

File: src/redux/header/tests/header.sagas.test.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        expect(analysisPut).toHaveLength(3);
+      });
+  });
+
+  it('should delete header from store', () => {
+    expectSaga(handleHeaderDelete, { payload: id })
+      .withReducer(combineReducers({ Header }), { Header: initialState })
+      .put(setHeaderLoading(true))
+      .provide([
+        [call(deleteHeader, id)],
```
Comment (ivan-bonk) [inline src/redux/header/tests/header.sagas.test.js:101]: here you call "deleteHeader" from actions
import this function from operations please

Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:101]: thanks

File: src/redux/header/tests/header.sagas.test.js (hunk 1 above)
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:26]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      .put(setHeaderLoading(true))
+      .provide([
+        [call(updateHeader, mockHeader), mockHeader],
+        [call(handleSuccessSnackbar, SUCCESS_UPDATE_STATUS)]
+      ])
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        expect(analysisPut).toHaveLength(6);
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:132]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  });
+
+  it('should update header', () => {
+    expectSaga(handleHeaderDelete, { payload: mockHeader })
+      .withReducer(combineReducers({ Header }), { Header: initialState })
+      .put(setHeaderLoading(true))
+      .provide([
+        [call(updateHeader, mockHeader), mockHeader],
+        [call(handleSuccessSnackbar, SUCCESS_UPDATE_STATUS)]
+      ])
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:127]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        expect(analysisPut).toHaveLength(3);
+      });
+  });
+
+  it('should update header', () => {
+    expectSaga(handleHeaderDelete, { payload: mockHeader })
+      .withReducer(combineReducers({ Header }), { Header: initialState })
+      .put(setHeaderLoading(true))
+      .provide([
+        [call(updateHeader, mockHeader), mockHeader],
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:125]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        expect(analysisPut).toHaveLength(3);
+      });
+  });
+
+  it('should update header', () => {
+    expectSaga(handleHeaderDelete, { payload: mockHeader })
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:121]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    expectSaga(handleHeaderLoad, { payload: id })
+      .withReducer(combineReducers({ Header }), { Header: initialState })
+      .put(setHeaderLoading(true))
+      .provide([[call(getHeaderById, id), mockHeader]])
+      .put(setHeader(mockHeader))
+      .put(setHeaderLoading(false))
+      .hasFinalState({
+        Header: {
+          ...initialState,
+          list: [],
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:67]: thanks

File: src/redux/header/tests/header.sagas.test.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+} from '../header.actions';
+
+const { SUCCESS_ADD_STATUS, SUCCESS_UPDATE_STATUS } = config.statuses;
+
+describe('header sagas tests', () => {
+  it('should load all headers', () => {
+    expectSaga(handleHeadersLoad)
+      .withReducer(combineReducers({ Header }), { Header: initialState })
+      .put(setHeaderLoading(true))
+      .provide([[call(getAllHeaders)], headers])
```
[the remark this answers is not in the document]
Comment (nadiyafomenko) [reply src/redux/header/tests/header.sagas.test.js:40]: thanks

**Review by moran711 — APPROVED** [review]

**Review by ivan-bonk — APPROVED** [review]

**Review by VitalikVoicix — APPROVED** [review]

**Pull request merged**